File: MergeSortedArr.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        ptr1=m-1
        ptr2=m+n-1
        ptr3=n-1
        while ptr1>=0 and ptr3>=0:
            if nums1[ptr1]>nums2[ptr3]:
                nums1[ptr2]=nums1[ptr1]
                ptr1-=1
            else:
                nums1[ptr2]=nums2[ptr3]
                ptr3-=1
            ptr2-=1
        while ptr3>=0:
            nums1[ptr2]=nums2[ptr3]
            ptr3-=1
            ptr2-=1
            
"""Time complexity-O(m+n)
Space Complexity - O(1)"""


# for i in range(m,len(nums1)):
#             nums1[i]=nums2[i-m]
#         nums1.sort()
        
#         """Time complexity-O(nlog(m+n)) as we are appending the nums2 array into nums1 and then sorting the whole nums1 array
#         Space Complexity - O(1)"""
        
        

File: mergeSortedArr.js
// Time Complexity : O(n)
// Space Complexity : O(1) - constant
// Did this code successfully run on Leetcode : yes

// We utilized 2 pointers to see where values were and compared each of them.
// We also had a check to verify that a pointer wasn’t undefined: pointer1.
// If it was, we would just update the value in the array.

var merge = function (nums1, m, nums2, n) {
  let pointer1 = m - 1;
  let pointer2 = n - 1;

  if (pointer2 < 0) {
    return nums1;
  }

  for (let i = nums1.length - 1; i >= 0; i--) {
    if (nums2[pointer2] >= nums1[pointer1] || nums1[pointer1] === undefined) {
      nums1[i] = nums2[pointer2];
      pointer2--;
    } else {
      nums1[i] = nums1[pointer1];
      pointer1--;
    }
  }
  return nums1;
};
